src/System.Runtime.Loader/tests/DefaultContext/DefaultLoadContextTest.cs

```diff
@@ -58,24 +58,43 @@ public static void LoadInDefaultContext()
         {
             Init();
 
-           // This will attempt to load an assembly, by path, in the Default Load context via the Resolving event
-           var assemblyName = "System.Runtime.Loader.Noop.Assembly";
+            // This will attempt to load an assembly, by path, in the Default Load context via the Resolving event
+            var assemblyNameStr = "System.Runtime.Loader.Noop.Assembly";
+            var assemblyName = new AssemblyName(assemblyNameStr);
 
+            // By default, the assembly should not be found in DefaultContext at all
+            Assert.Throws(typeof(FileNotFoundException), () => Assembly.Load(assemblyName));
+
+            // Now, wireup the Resolving event to locate the assembly
             AssemblyLoadContext.Default.Resolving += ResolveAssembly;
-            var assemblyExpected = AssemblyLoadContext.Default.LoadFromAssemblyName(new AssemblyName(assemblyName));
+            
+            // This will invoke the resolution via VM requiring to bind using the TPA binder
+            var assemblyExpectedFromLoad = Assembly.Load(assemblyName);
+
+            // We should have successfully loaded the assembly in default context.
+            Assert.NotNull(assemblyExpectedFromLoad);
+
+            // And make sure the simple name matches
+            Assert.Equal(assemblyNameStr, assemblyExpectedFromLoad.GetName().Name);
+
+            // This will resolve the assembly via event invocation.
+            var assemblyExpected = AssemblyLoadContext.Default.LoadFromAssemblyName(assemblyName);
             
             // We should have successfully loaded the assembly in default context.
             Assert.NotNull(assemblyExpected);
 
+            // What we got via Assembly.Load and LoadFromAssemblyName should be the same
+            Assert.Equal(assemblyExpected, assemblyExpectedFromLoad);
+
             // And make sure the simple name matches
-            Assert.Equal(assemblyExpected.GetName().Name, assemblyName);
+            Assert.Equal(assemblyExpected.GetName().Name, assemblyNameStr);
 
             // Unwire the Resolving event.
             AssemblyLoadContext.Default.Resolving -= ResolveAssembly;
```

Comment:
> unless you're saying that every invocation returns the results in random order

No, am not saying that, though the order of GetInvocationList is also an implementation detail of the runtime.

> This wouldn't be as big an issue if this was a normal void-returning event handler

If the code were to take dependency on the order of callbacks, then even with void returning handler it could create a problem if the app modified its state based upon its assumption of what all callbacks may have been already invoked.

> making the ordering very observable and impacting behavior.

This is not representative of registration ordering.

> If it were my code, I'd add tests for this.

I have thought about this - I think a better test would be to ensure that we only invoke a non-Null returning callback once, as opposed to validating the callback returning non-Null corresponding to its registration order (was it the first one we registered or the second and so on). This should help validate the invariant of "break on first non-Null" in Resolving event independent of the registration/invocation order.